rigex-front-end/src/components/createWellModal/createWell.js

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect } from 'react';
+import { useMutation } from '@apollo/client';
+import { useDispatch } from 'react-redux';
+import Dialog from "@material-ui/core/Dialog";
+import CloseIcon from "@material-ui/icons/Close";
+
+import { wellActions } from '../../store/wells/action';
+import { UPSERT_WELL } from '../../constants/serviceAPI';
+import "./wellModalStyles.scss"
+
+export default function CreateWellModal({wellModalOpen, handleWellModalClose, rigId}) {
+
+    const [wellInfo, setWellInfo] = useState({
+        name: "",
+        latitude: "",
+        longitude: ""
+    });
+    const [upsertWell, { loading, error, data }] = useMutation(UPSERT_WELL);
+    const dispatch = useDispatch();
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+
+        await upsertWell({
+            variables: {wellInput: {
+                name: wellInfo.name,
+                longitude: parseFloat(wellInfo.longitude),
+                latitude: parseFloat(wellInfo.latitude),
+                rigId: rigId.toString()
+            }}
+        })
+
+        // check state for no errors, then close modal or reset wellInfo
+    }
+
+    useEffect(() => {
+      if (data) {
+          dispatch(wellActions.upsertWell(data.upsertWell))
+      }
+    }, [dispatch, loading, error, data])
+
+    const [validName, setValidName] = useState(false)
+
+    const re = /^-?[0-9]{1,3}(?:\.([0-9]{1,10})?)?$/
+    const handleChange = (event) => {
+
+        let name = event.target.name
+        console.log(name)
+
+        // prevent typing non-integer values or decimal into non-name fields
```

Comment:
One way to think of code is that it should be readable. Now just cause it's not in english doesn't mean we shouldn't be able to read code the same as we read a book. Your code should be written in a way that is clear enough that you do not need to tell other programmers what your code does in english. This violates the dry principle.